torch._dynamo.reset() before and after running each test

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110107

Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

zou3519 committed:
torch._dynamo.reset() before and after running each test

Test Plan:
- CI

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

[ghstack-poisoned]

voznesenskym commented on torch/testing/_internal/common_utils.py:
```diff
@@ -2467,14 +2467,25 @@ def _run_with_retry(self, result=None, num_runs_left=0, report_only=True, num_re
+            def inner(*args, **kwargs):
+                # It's important to reset the Dynamo state so that previous
+                # invocations do not affect future ones.
+                torch._dynamo.reset()
+                try:
+                    return torch._dynamo.optimize(backend)(function)(*args, **kwargs)
+                finally:
+                    torch._dynamo.reset()
```
Why after too?

zou3519 replied:
It's good to clean up after ourselves, but mechanically we don't really need it

voznesenskym replied:
ok

ezyang reviewed (approved):
You can ship this if it doesn't cause timeouts 🤔

zou3519 commented:
> You can ship this if it doesn't cause timeouts 🤔

I've solved the timeouts problem with more shards. The problem right now is skipping enough tests until the CI is green lol

malfet reviewed (approved):
Let's keep an eye on potentially increasing the queue time as we are using more runners now

malfet commented:
> I've solved the timeouts problem with more shards. The problem right now is skipping enough tests until the CI is green lol

How much longer it takes with cache invalidation then without it?

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 commented:
> > I've solved the timeouts problem with more shards. The problem right now is skipping enough tests until the CI is green lol
> 
> How much longer it takes with cache invalidation then without it?

The total time is now 2-3x of what it was previously. So far I have increased the shard count from 2 to 5

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "torch._dynamo.reset() before and after running each test"


Previously, running tests with PYTORCH_TEST_WITH_DYNAMO=1 ran some tests in eager-mode pytorch because the Dynamo cache filled up. We need to reset the Dynamo state before each test in order to actually run each test using Dynamo instead of non-deterministically falling back to eager-mode.

Test Plan:
- CI

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]
